runtime/gc_glue_java/ObjectModel.hpp

```diff
@@ -562,6 +562,20 @@ class GC_ObjectModel : public GC_ObjectModelBase
 	{
 		GC_ObjectModelBase::fixupForwardedObject(forwardedHeader, destinationObjectPtr, objectAge);
 
+#if defined(J9VM_ENV_DATA64)
+		if (_javaVM->isIndexableDataAddrPresent) {
```

Comment:
Hm, why do we need this!? This path is only taken by standard GCs (that will soon not have dataAddr present). Balanced already invokes fixupDataAddr at its specific spots that do object movement/require fixup.

Reply:
Correct, however, this is needed due to the macro access changes in `j9accessbarrier.h`, which rely on `dataAddr` if `dataAddr` is present. Once `dataAddr` is never present for standard collectors and option `-XXgc:disableIndexableDualHeaderShape` is removed, this will no longer be needed.